imod/protocols/protocol_fiducialModel.py

```diff
@@ -0,0 +1,811 @@
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+
+import os
+import numpy as np
+import pyworkflow as pw
+import pyworkflow.em as pyem
+import pyworkflow.protocol.params as params
+import pyworkflow.em.data as data
+import tomo.objects as tomoObj
+from tomo.objects import LandmarkModel
+from tomo.protocols import ProtTomoBase
+from tomo.convert import writeTiStack
+
+
+class ProtFiducialModel(pyem.EMProtocol, ProtTomoBase):
+    """
+    Construction of a fiducial model based on the IMOD procedure.
+    More info:
+        https://bio3D.colorado.edu/imod/doc/etomoTutorial.html
+    """
+
+    _label = 'fiducial model'
+
+    # -------------------------- DEFINE param functions -----------------------
+    def _defineParams(self, form):
+        form.addSection('Input')
+        form.addParam('inputSetOfTiltSeries',
+                      params.PointerParam,
+                      pointerClass='SetOfTiltSeries',
+                      important=True,
+                      label='Input set of tilt-Series')
+
+        form.addParam('twoSurfaces',
+                      params.EnumParam,
+                      choices=['Yes', 'No'],
+                      default=0,
+                      label='Find on two surfaces',
+                      important=True,
+                      display=params.EnumParam.DISPLAY_HLIST,
+                      help="Track fiducials differentiating in which side of the sample are located.")
+
+        form.addParam('fiducialDiameter',
+                      params.FloatParam,
+                      label='Fiducial diameter',
+                      default='4.95',
+                      help="Fiducials diameter to be tracked for alignment.")
+
+        form.addParam('numberFiducial',
+                      params.IntParam,
+                      label='Number of fiducials',
+                      default='25',
+                      expertLevel=params.LEVEL_ADVANCED,
+                      help="Number of fiducials to be tracked for alignment.")
+
+        form.addParam('rotationAngle',
+                      params.FloatParam,
+                      label='Tilt rotation angle (deg)',
+                      default='0.0',
+                      expertLevel=params.LEVEL_ADVANCED,
+                      help="Angle from the vertical to the tilt axis in raw images.")
+
+        form.addParam('importTrackFile',
+                      params.EnumParam,
+                      choices=['Yes', 'No'],
+                      default=1,
+                      label='Import track file',
+                      important=True,
+                      display=params.EnumParam.DISPLAY_HLIST,
+                      expertLevel=params.LEVEL_ADVANCED,
+                      help="Import a customized track file for the execution of the program.")
+
+        groupTrackFile = form.addGroup('Track file',
```

Comment:
What is a track file...is it like a reference?

Reply:
Track file is the configuration file generated by Etomo in order to run the program Beadtrack for fiducial tracking through the tilt images of the series.